File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/domain/converter/OzToLbConverterTest.java
package com.tarbadev.witchcraft.domain.converter;

import com.tarbadev.witchcraft.domain.converter.OzToLbConverter;
import org.junit.Before;
import org.junit.Test;

import static org.assertj.core.api.Java6Assertions.assertThat;

public class OzToLbConverterTest {
  private OzToLbConverter ozToLbConverter;

  @Before
  public void setUp() {
    ozToLbConverter = new OzToLbConverter();
  }

  @Test
  public void convert() {
    Double oz = 16.0;
    Double expectedLb = 1.0;

    assertThat(ozToLbConverter.convert(oz)).isEqualTo(expectedLb);

    Double oz2 = 20.0;
    Double expectedLb2 = 1.25;

    assertThat(ozToLbConverter.convert(oz2)).isEqualTo(expectedLb2);

    Double oz3 = 5.0;
    Double expectedLb3 = 0.3125;

    assertThat(ozToLbConverter.convert(oz3)).isEqualTo(expectedLb3);
  }
}

File: WitchcraftApp/src/test/kotlin/com/tarbadev/witchcraft/converter/OzToLbConverterTest.kt
package com.tarbadev.witchcraft.converter

import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.assertEquals

class OzToLbConverterTest {
    private lateinit var ozToLbConverter: OzToLbConverter

    @BeforeEach
    fun setUp() {
        ozToLbConverter = OzToLbConverter()
    }

    @Test
    fun convert() {
        val oz = 16.0
        val expectedLb = 1.0

        assertEquals(expectedLb, ozToLbConverter.convert(oz))

        val oz2 = 20.0
        val expectedLb2 = 1.25

        assertEquals(expectedLb2, ozToLbConverter.convert(oz2))

        val oz3 = 5.0
        val expectedLb3 = 0.3125

        assertEquals(expectedLb3, ozToLbConverter.convert(oz3))
    }
}